Assets/LeapMotion/Modules/GraphicRenderer/Scripts/Graphics/ProceduralMeshGraphics/LeapSlicedGraphic.cs

```diff
@@ -0,0 +1,389 @@
+    /// Resolution along the X and Y axes are provided, as well as mesh-space and 
+    /// UV-space margins to pass into calculateVertAxis as border0 and border1 to support
+    /// nine slicing (see Mesh Data Support).
+    /// </summary>
+    public abstract void RefreshSlicedMeshData(Vector2i    resolution,
+                                               RectMargins meshMargins,
+                                               RectMargins uvMargins);
+
+    #region Mesh Data Support
+
+    /// <summary>
+    /// Given a vertex index from an edge, the total vertCount and size
+    /// along the current dimension, and the distance to the beginning and end borders
+    /// along the current dimension (if nine-slicing is enabled ONLY), this method will
+    /// return the distance along the current dimension that the vertIdx should be from
+    /// the origin of the dimension.
+    /// 
+    /// If the sliced graphic has nine-slicing enabled, or if alwaysRespectBorder is 
+    /// set to true, the second and second-to-last vertices along an edge will reflect
+    /// the border margin arguments instead of a uniform tesselation along the whole
+    /// width of the dimension. Inner vertices will then respect this non-uniformity,
+    /// slicing the rest of the inner width evenly.
+    /// 
+    /// e.g. Determine the distance along the X axis for:
+    /// - vertIdx = 4, the fifth vertex along the X axis
+    /// - vertCount = 16, given sixteen total grid columns (16 verts per X-axis row)
+    /// - size = 2, given a box width of 2 units
+    /// - border0 = 0.2f, given the left border starts at 0.2 units from the left edge
+    /// - border1 = 0.2f, given the right border starts at 0.2 units from the right edge
+    /// This method will return the X axis coordinate of the fifth vertex accordingly.
+    /// To calculate the corresponding Y axis coordinate, simply provide the vertCount,
+    /// size, and border information for the Y dimension instead.
+    /// </summary>
+    protected float calculateVertAxis(int vertIdx, int vertCount, float size, float border0, float border1, bool alwaysRespectBorder = false) {
+      if (_nineSliced || alwaysRespectBorder) {
+        if (vertIdx == 0) {
+          return 0;
+        }
+        else if (vertIdx == (vertCount - 1)) {
+          return size;
+        }
+        else if (vertIdx == 1) {
+          return border0;
+        }
+        else if (vertIdx == (vertCount - 2)) {
+          return size - border1;
+        }
+        else {
+          return ((vertIdx - 1.0f) / (vertCount - 3.0f)) * (size - border0 - border1) + border0;
+        }
+      }
+      else {
+        return (vertIdx / (vertCount - 1.0f)) * size;
+      }
+    }
+
+    #endregion
+
+    #region Supporting Types
+
+    public enum ResolutionType {
+      Vertices,
+      VerticesPerRectilinearMeter
+    }
+
+    public struct Vector2i {
+      public int x;
+      public int y;
+    }
+    
+    public struct RectMargins {
```

Comment:
Yeah, you can't set left/right/top/bottom directly in a Rect struct, and the meaning of right and bottom is different in a Rect struct.

Right and Bottom in RectMargins are measured from the right and bottom edges respectively, instead of referring to the distance along the relevant axis _to_ the right and bottom edges (In UV space, 1 - right or 1 - bottom). I added some documentation to make that more clear.